File: backend/src/main/java/de/rubeen/bsc/entities/web/CalendarEntity.java
package de.rubeen.bsc.entities.web;

import com.google.api.services.calendar.model.CalendarListEntry;
import de.rubeen.bsc.entities.db.tables.records.CalendarRecord;
import de.rubeen.bsc.provider.office365.entities.Calendar;

public class CalendarEntity {
    private String calendarName, calendarID;
    private boolean activated;
    private String provider;

    public CalendarEntity() {}

    public CalendarEntity(CalendarListEntry calendarListEntry, boolean activated) {
        this.calendarName = calendarListEntry.getSummary();
        this.calendarID = calendarListEntry.getId();
        this.activated = activated;
        this.provider = "google";
    }

    public CalendarEntity(String calendarName, String calendarID, boolean activated, String provider) {
        this.calendarName = calendarName;
        this.calendarID = calendarID;
        this.activated = activated;
        this.provider = provider;
    }

    public CalendarEntity(Calendar calendar, boolean isActivated) {
        this.calendarName = calendar.getName();
        this.calendarID = calendar.getId();
        this.activated = isActivated;
        this.provider = "office";
    }

    public String getCalendarName() {
        return calendarName;
    }

    public String getCalendarID() {
        return calendarID;
    }

    public boolean isActivated() {
        return activated;
    }

    public String getProvider() {
        return provider;
    }

    public void setProvider(String provider) {
        this.provider = provider;
    }
}

File: backend/src/main/java/de/rubeen/bsc/entities/web/CalendarEntity.kt
package de.rubeen.bsc.entities.web

import com.google.api.services.calendar.model.CalendarListEntry
import de.rubeen.bsc.provider.office365.entities.Calendar

class CalendarEntity {
    var calendarName: String? = null
        private set
    var calendarID: String? = null
        private set
    var activated: Boolean = false
        private set
    var provider: String? = null
        private set

    constructor()

    constructor(calendarListEntry: CalendarListEntry, activated: Boolean) {
        this.calendarName = calendarListEntry.summary
        this.calendarID = calendarListEntry.id
        this.activated = activated
        this.provider = "google"
    }

    constructor(calendarName: String, calendarID: String, activated: Boolean, provider: String) {
        this.calendarName = calendarName
        this.calendarID = calendarID
        this.activated = activated
        this.provider = provider
    }

    constructor(calendar: Calendar, isActivated: Boolean) {
        this.calendarName = calendar.name
        this.calendarID = calendar.id
        this.activated = isActivated
        this.provider = "office"
    }
}
